RELEASING.md

```diff
@@ -0,0 +1,63 @@
+# Releasing Kubeclient
+
+## Versioning
+Kubeclient release versioning follows [SemVer](https://semver.org/).
+At some point in time it is decided to release version x.y.z.
+Prerequisite: A [changelog](CHANGELOG.md) is generated & merged into the release branch.
+
+```bash
+RUBYGEMS_USERNAME=bob
+RELEASE_BRANCH="master"
+RELEASE_VERSION="x.y.z"
+GIT_REMOTE="origin"
+GIT_UPSTREAM="upstream"
+
+#
+# Install the release gem
+#
+gem install gem-release
+
+
+#
+# Checking out the release branch
+#
+git fetch $GIT_UPSTREAM
+git checkout $GIT_UPSTREAM/$RELEASE_BRANCH -b $RELEASE_BRANCH
+
+#
+# Preparing to release
+#
+bundle install
+bundle exec rake test rubocop
+
+#
+# Grabbing an authentication token for rubygems.org api
+#
+curl -u $RUBYGEMS_USERNAME https://rubygems.org/api/v1/api_key.yaml > ~/.gem/credentials; chmod 0600 ~/.gem/credentials
+
+cat ~/.gem/credentials
+---
+:rubygems_api_key: ****
+
+#
+# Bumping the gem's version can be done manually, or by using
+#
+gem bump --version $RELEASE_VERSION --push --destination $GIT_REMOTE
+
+# ... Create & Merge A version bump pull request
```

Comment:
if we're doing bump via PR, should `git pull $GIT_UPSTREAM` after merging to tag correctly.
that also rules out the all-in-one `gem bump --version $RELEASE_VERSION --destination $GIT_REMOTE --tag --release`.
(I'm ok with either PR or direct bump push, but let's decide & document just one flow...)